parlai/core/torch_agent.py

```diff
@@ -993,10 +1059,16 @@ def observe(self, observation):
 
         This includes remembering the past history of the conversation.
         """
+        obs = observation.copy()
+
+        if 'context' in obs and not self.opt.get('parse_context', False):
+            # default to merging context into text, not parsing it separately
+            obs['text'] = obs.pop('context') + '\n' + obs['text']
```

Comment:
brain dump:

Should we do the .copy here instead?

As I recall, one advantage of NOT copying the data here is that the vectorized text is saved in case we visit this observation again. (We actually go through great lengths in `_set_text_vec` etc to do this). In fact, I'm not even sure that pytorch data teacher doesn't rely on that (admittedly devious) behavior -- I'd have to check.

It gets real weird because we're also going to start adding p1 after newlines, which might get into the context? Though I guess we're already doing that, aren't we?